### ISSUE TITLE
Regular Allocating and checking while seeding 

### ISSUE THREAD

**Issue by CptVinyl** [BiglySoftware/BiglyBT#2128]
Hello. 

Now new strange BiglyBT behavior. 
While seeding big torrent (~36500 files, 110 GB) at random moments seeding interrupted with "Allocating":
![browser_hvGFxTwgcf](https://user-images.githubusercontent.com/84813181/121737571-b8288080-cb01-11eb-910a-63177537b4b7.png)
![browser_qulmO6BxHF](https://user-images.githubusercontent.com/84813181/121737589-bf4f8e80-cb01-11eb-8392-e3932d6928a5.png)

After "Allocating" ended (with normal progression), torrent became with status "Checking 0.00%", stay with 0.00% near 15-30 minutes, no progression, no errors:
![browser_q8ZYcVsCdT](https://user-images.githubusercontent.com/84813181/121737653-d0000480-cb01-11eb-984f-0bf4cb7ebae4.png)
![browser_xY1VavckUf](https://user-images.githubusercontent.com/84813181/121737679-d7271280-cb01-11eb-9d79-1a072b52b47b.png)

And then starts normally. Seeding for 1-3 hours and then again "Allocating".
This happens to all torrents bigger than 50 GB with active seeding (continuously for 1-3 hours, on high speed 4+MiB\s).
The torrents were successfully downloaded, files are good, no corrupted data. 

So the question.
Is this normal activity for BiglyBT and I shouldn't worry about this? Or I'm doing something wrong?
Maybe bad configuration \ settings?

I test it on JAVA 13, 14, 15, 16. This behavior is actual on all of them. 

I`m not using any sync. software such as dropbox, google drive or smth.
Just normal download and seeding on hard drive. Other software do not touching this disk. 

Java 15.0.2 (64 bit)
  Oracle Corporation

SWT v4942r22, win32, zoom=100, dpi=96
Windows 10 v10.0, amd64 (64 bit)
B2.7.0.3_B26/4 az3

**Comment by parg** [comment]
Torrents naturally go through allocating and checking when starting up (from queued for example) - are you sure your torrents aren't simply switching between seeding+queued based on whatever seeding rules you have setup?

**Comment by CptVinyl** [comment]
Hm, that's interesting. Need to investigate this for a few days. I will report the results later.

Do you understand you correct?
You mean, that it is normal if torrent became "queued" for... lets say 1-2 minutes, and than starting up again with allocating and checking? Every time when seeding starts.

**Comment by parg** [comment]
Yes - shouldn't get stuck 'checking' for any period of time though unless other downloads are actually checking themselves.

**Comment by parg** [comment]
(during the 'allocating' state BiglyBT is verifying that all the download's files exist and are of the expected size, so if you have a download with a large number of files this can take a short while)

**Comment by CptVinyl** [comment]
Well here my results. 

Yes, it became "Allocating" after queued. No problem here. Now I understand why, thanks for explanation. 

But what a strange thing I have discovered. 
The torrent is seeding on good speed, connected to peers and at random time it switches to queued and immediately starts again. (this all happened in ~1 minute).
Ok, maybe there is my fault in seeding settings, never mind. 

Torrent goes through "Allocating" stuck on Checking "0.00%" for at least 15 minutes. 
And after "Stucked Checking" the torrent starts seeding. 

So the most important question for me is why checking has no progression?
Stuck on 0.00% and after 15+ minutes starts?
There is no other disk activity at the same time:
![BiglyBT_JBUIkRuVjx](https://user-images.githubusercontent.com/84813181/122011644-fdc8a000-cdc4-11eb-9f75-806547cea9b9.png)
Also, at library, no other torrents with "Allocating" or "Checking" status:
![BiglyBT_WPVqh6ixRH](https://user-images.githubusercontent.com/84813181/122012844-4765ba80-cdc6-11eb-9d26-79a104119890.png)

Can the size of the torrent be a reason? Or the number of files?
I can not se the same "activity" on other torrents. They became active right after "Allocating" no stucked checking.

**Comment by parg** [comment]
That is pretty weird. If you can catch the torrent in a 'queued' state (before it goes into 'checking') what does the 'seeding rank' column show? (the one that shows 'first priority 2233' in your screenshot) Also try hovering your mouse over the seeding rank details to show more information in a tooltip (might need to enable 'Options->Queue: log debug information' to get the tooltip to display - try it now and check)

**Comment by CptVinyl** [comment]
I can not catch the torrent in a 'queued' state, because it happens in 0.5-1 second.
Without your prompting, I didn't even notice that torrent was becoming inactive. 
Not enough time to catch this with mouse :( 

Maybe there is a file with a log of seeding rank? I could look there.

While allocating seeding rank show this:
![BiglyBT_41TOmdQWvb](https://user-images.githubusercontent.com/84813181/122031804-b7307100-cdd7-11eb-8f1d-b4a54b5415c8.png)

**Comment by parg** [comment]
With that debug option enabled information gets logged to a view but not a file - see Tools->Plugin->Log Views->Start/Stop Rules

Unfortunately if you have a lot of active downloads the view gets pretty useless :(

**Comment by CptVinyl** [comment]
Let's try to stop all torrents except this one and look what will be in the view :)

**Comment by CptVinyl** [comment]
That is how the "sop\start" rules look like while checking is stucked at 0.00% for 15 minutes. 
Also I marked the place with an arrow, when status from checking 0.00% became "Seeding":
![BiglyBT_evhq6ECzGv](https://user-images.githubusercontent.com/84813181/122036679-80a92500-cddc-11eb-875f-dbeb896a7ba1.png) 

Now I will be waiting for "queued"

**Comment by parg** [comment]
Seems that it actually was doing 'something' for all that time when displaying 'checking 0,00%' - I wonder what :(

The trace shows the download sitting in state 'P' (preparing = combination of allocating+checking) and when it finally became 'R' (ready) things started working, 'R' is the state that a download transitions to when it has completed the checking process.

**Comment by parg** [comment]
If you manually stop the downoad and then start it does it go into the 'checking 0,0%' state for a long time or does it start seeding?

**Comment by CptVinyl** [comment]
> If you manually stop the download and then start it does it go into the 'checking 0,0%' state for a long time or does it start seeding?

Yes, If I manually stop the download and then start it goes through all the process:
- Allocating
- Checking 0.0% (for ~ minutes)
- Start Seeding
It does not start seeding immediately after launch, stay at Checking 0.0% status. 

It looks like it actually checking something, but not displaying progress.

**Comment by parg** [comment]
Here's something you could try - go to Options->Files and **enable** the option named

`Defer completed download file checks until first use - ...`

See if that removes the delay when starting the download.

**Comment by CptVinyl** [comment]
Need to test this for a few days.

Because now I can say that you are the best!
This option solved the problem completely! No issue for last few hours (after enabling the option, the problem did not appear).
Now it starts in seconds! 1-3 seconds of Allocating (at 0.0% btw), no any other delays. 
Nice! 

What a magic did you do here? :)

I think... Issue closed?

**Comment by NaPurHab** [comment]
@parg, I only just read this thread.  I can confirm that very large torrents that were started with the `Allocate then Queue` option in the download dialog. Sometimes immediately after that First Allocate, then Check. Seems redundant. I thought this issue was dealt with long ago.  But not only that, but I have then seen these large torrents, either allocate or check these torrents at random, that were allocated when added.   This is also NEW recent behaviour.  I regularly download torrents that are  10gb or larger, largest I have at the moment is 112gb.   But I have only seen this issue with large torrents in the last few betas, did not really think about the issue until I read this thread. So, I would say no farther back than 2.7.0.3_B24, I'm currently on 2.7.0.3_B26. 

I am wondering if that `Defer completed download file checks until first use - ...` , should not be something like 'Defer completed download file checks until torrent complete - ...'. So instead of after each file, only after the torrent is complete. And something else to consider, if the `Move on Complete` is set to a different file system, perhaps checking should be deferred until the torrent is in its final location. If 'Move on Complete' is to the same file system, it is just a retarget.

**Comment by parg** [comment]
Don't think there;s any recent changes that might have affected this,

When the download is starting up one of the checks that BiglyBT runs is to get the size of every file in the download on disk and verify that this is as expected. If you have thousands of files then this will require thousands of disk accesses and if the disk is slow it seems that can cause significant delay,

That option disables this check - the disadvantage is that if a file is the wrong size then that will trigger an error later when the download is running.

**Comment by CptVinyl** [comment]
I think issue closed? 
As discovered, no issue here, just need to disable "Defer completed download file checks"

Thanks :)

**Comment by parg** [comment]
I'll fiddle with it so at least the percentage increases rather than being stuck at 0%

**Comment by CptVinyl** [comment]
Thanks. I am very grateful for your help and the work you are done! :)
